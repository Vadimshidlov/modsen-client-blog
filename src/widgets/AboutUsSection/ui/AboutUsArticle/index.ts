export { AboutUsArticle } from '@/widgets/AboutUsSection/ui/AboutUsArticle/AboutUsArticle';
