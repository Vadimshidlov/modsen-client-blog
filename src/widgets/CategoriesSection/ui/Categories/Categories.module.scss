.categoriesContainer {
  margin-bottom: 48px;

  .categoriesTitle {
    @include headingTwo;
    margin-bottom: 40px;

    @media (max-width: $tablet) {
      margin-bottom: 20px;
      @include headingFour;
    }

    @media (max-width: $mobile) {
      margin-bottom: 20px;
      @include headingFour;
    }
  }

  .categoriesListContainer {
    display: flex;
    flex-direction: column;
    row-gap: 24px;

    @media (max-width: 1250px) {
      flex-direction: row;
      gap: 10px;
      // justify-content: space-between;
      flex-wrap: wrap;
    }

    @media (max-width: 1000px) {
      // justify-content: center;
      gap: 10px;
      justify-content: start;
    }

    @media (max-width: 768px) {
      flex-direction: column;
    }

    .categoriesItem {
      display: flex;
      column-gap: 16px;
      padding: 24px;
      border: 1px solid $grey;
      cursor: pointer;
      // max-width: 296px;
      width: 296px;
      // width: 100%;
      align-items: center;

      &:hover {
        background: $yellow;
      }

      @media (max-width: 1000px) {
        padding: 12px;
      }

      @media (max-width: 768px) {
        width: 100%;
      }

      .categoriesItemImage {
      }

      .categoriesItemTitle {
        @include headingThree;

        @media (max-width: 1250px) {
          @include headingFive;
        }
      }
    }
  }
}
