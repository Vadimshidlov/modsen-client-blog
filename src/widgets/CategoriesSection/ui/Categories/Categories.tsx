import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import { useTranslations } from 'next-intl';
import { CATEGORIES } from '@/shared/constants/categories';
import styles from '@/widgets/CategoriesSection/ui/Categories/Categories.module.scss';

export function Categories() {
  const t = useTranslations('Category');

  return (
    <div className={styles.categoriesContainer}>
      <h2 className={styles.categoriesTitle}>{t('title')}</h2>
      <ul className={styles.categoriesListContainer}>
        {CATEGORIES.map(({ title, img }) => (
          <Link key={title} href={`/category/${title.toLowerCase()}`}>
            <li className={styles.categoriesItem}>
              <Image className={styles.categoriesItemImage} src={img} alt="owerview-image" />
              <span className={styles.categoriesItemTitle}>{t(`${title}`)}</span>
            </li>
          </Link>
        ))}
      </ul>
    </div>
  );
}
