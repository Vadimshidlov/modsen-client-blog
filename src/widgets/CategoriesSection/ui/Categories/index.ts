export { Categories } from '@/widgets/CategoriesSection/ui/Categories/Categories';
