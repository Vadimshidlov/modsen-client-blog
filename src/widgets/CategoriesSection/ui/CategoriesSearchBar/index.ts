export { CategoriesSearchBar } from '@/widgets/CategoriesSection/ui/CategoriesSearchBar/CategoriesSearchBar';
