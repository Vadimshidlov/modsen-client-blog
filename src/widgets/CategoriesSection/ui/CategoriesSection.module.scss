.categoriesSection {
  display: flex;
  margin-bottom: 128px;
  padding-top: 102px;
  // column-gap: 147px;
  column-gap: 40px;
  justify-content: space-between;

  @media (max-width: 1250px) {
    flex-direction: column-reverse;
  }

  @media (max-width: $tablet) {
    padding-top: 51px;
    margin-bottom: 64px;
  }

  @media (max-width: $mobile) {
    padding-top: 25px;
    margin-bottom: 32px;
  }

  .postsContainer {
    display: flex;
    flex-direction: column;
    row-gap: 32px;
    max-width: 844px;

    .postsNotFountTitle {
      @include headingThree;

      @media (max-width: $tablet) {
        @include headingThree;
      }

      @media (max-width: $mobile) {
        @include headingFive;
      }
    }
  }

  .searchBarWrapper {
    max-width: 300px;
    display: flex;
    flex-direction: column;

    @media (max-width: 1250px) {
      max-width: 100%;
    }
  }
}
