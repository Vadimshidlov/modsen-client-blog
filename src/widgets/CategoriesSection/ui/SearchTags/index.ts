export { SearchTags } from '@/widgets/CategoriesSection/ui/SearchTags/SearchTags';
