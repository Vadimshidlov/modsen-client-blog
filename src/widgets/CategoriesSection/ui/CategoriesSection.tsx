'use client';

import { useState } from 'react';
import styles from '@/widgets/CategoriesSection/ui/CategoriesSection.module.scss';
import container from '@/shared/styles/container.module.scss';
import { CategoriesSearchBar } from '@/widgets/CategoriesSection/ui/CategoriesSearchBar';
import { POSTS } from '@/shared/constants/posts';
import { PostItem } from '@/widgets/PostItem';
import { Categories } from '@/widgets/CategoriesSection/ui/Categories';
import { SearchTags } from '@/widgets/CategoriesSection/ui/SearchTags';
import { useTranslations } from 'next-intl';
import { CategoriesSectionPropsType } from '@/shared/types';

export function CategoriesSection({ selectedCategory }: CategoriesSectionPropsType) {
  const t = useTranslations('CategoriesSection');

  const [selectTags, setSelectTags] = useState<string[]>([]);

  const filteredPosts = POSTS.filter(
    (postData) =>
      postData.category.toLowerCase() === selectedCategory.toLowerCase() &&
      (selectTags.length === 0 || postData.tags.some((tag) => selectTags.includes(tag))),
  );

  return (
    <section className={`${styles.categoriesSection} ${container.container}`}>
      <div className={styles.postsContainer}>
        {filteredPosts.length === 0 && (
          <span className={styles.postsNotFountTitle}>{t('notFoundMessage')}</span>
        )}
        {filteredPosts.length !== 0 &&
          filteredPosts
            .slice(0, 3)
            .map(({ id, img, category, title, text }) => (
              <PostItem category={category} id={id} img={img} text={text} title={title} key={id} />
            ))}
      </div>
      <div className={styles.searchBarWrapper}>
        <SearchTags selectTags={selectTags} setSelectTags={setSelectTags} />
        <CategoriesSearchBar />
        <Categories />
      </div>
    </section>
  );
}
