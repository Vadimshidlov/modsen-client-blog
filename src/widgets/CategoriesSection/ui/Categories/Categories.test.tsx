import React from 'react';
import { render } from '@testing-library/react';
import { Categories } from '@/widgets/CategoriesSection/ui/Categories';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('Categories tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<Categories />);
    expect(container).toMatchSnapshot();
  });
});
