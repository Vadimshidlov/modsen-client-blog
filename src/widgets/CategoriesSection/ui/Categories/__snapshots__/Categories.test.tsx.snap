// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories tests should matches snapshot 1`] = `
<div>
  <div
    class="categoriesContainer"
  >
    <h2
      class="categoriesTitle"
    >
      Mocked title
    </h2>
    <ul
      class="categoriesListContainer"
    >
      <a
        href="/category/business"
      >
        <li
          class="categoriesItem"
        >
          <img
            alt="owerview-image"
            class="categoriesItemImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            class="categoriesItemTitle"
          >
            Mocked Business
          </span>
        </li>
      </a>
      <a
        href="/category/startup"
      >
        <li
          class="categoriesItem"
        >
          <img
            alt="owerview-image"
            class="categoriesItemImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            class="categoriesItemTitle"
          >
            Mocked Startup
          </span>
        </li>
      </a>
      <a
        href="/category/economy"
      >
        <li
          class="categoriesItem"
        >
          <img
            alt="owerview-image"
            class="categoriesItemImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            class="categoriesItemTitle"
          >
            Mocked Economy
          </span>
        </li>
      </a>
      <a
        href="/category/technology"
      >
        <li
          class="categoriesItem"
        >
          <img
            alt="owerview-image"
            class="categoriesItemImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <span
            class="categoriesItemTitle"
          >
            Mocked Technology
          </span>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
